apps/system/lockscreen/js/charging.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+(function(exports) {
+
+/**
+ * Creates an object used for refreshing the charging string UI element.
+ * @class  ChargingStatus
+ */
+function ChargingStatus() {
+  this.refreshCB = null;
+}
+
+ChargingStatus.prototype.start = function cs_start() {
+  this.refresh();
+
+  if (!this.refreshCB) {
+    this.refreshCB = this.refresh.bind(this);
+
+    navigator.battery.addEventListener('chargingchange',
+      this.refreshCB);
```

Comment:
And you can make it more neat if you set this in constructor:

    this.listenEvents = [...] // String names of these events

Then write this method as:

    this.listenEvents.forEach((ename) => { navigator.battery.addEventListener(ename, this); });